src/back/spv/writer.rs

```diff
@@ -439,6 +441,51 @@ impl Writer {
             results: Vec::new(),
         };
 
+        for (handle, expression) in ir_function.expressions.iter() {
+            match *expression {
+                crate::Expression::Compose { ty, .. } => {
+                    if let crate::TypeInner::Array { .. } = ir_module.types[ty].inner {
+                        let id = self.id_gen.next();
+                        let pointer_type_id = self.get_pointer_id(
+                            &ir_module.types,
+                            ty,
+                            spirv::StorageClass::Function,
+                        )?;
+                        let instruction = Instruction::variable(
+                            pointer_type_id,
+                            id,
+                            spirv::StorageClass::Function,
+                            None,
+                        );
+                        prelude.body.push(instruction);
+                        function.local_const_arrays.insert(handle, id);
+                    }
+                }
+                crate::Expression::Constant(h) => match ir_module.constants[h].inner {
```

Comment:
We probably want to check that `h` isn't already registered. I don't think there is a rule in the IR that prevents multiple `Expression::Constant` pointing to the same thing.